File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float change;
    int counter = 0;
    
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);
    // get user input for what is the change
    
    
    int amountLeft = round(change * 100);    // input amount is turned into cents

    while (amountLeft >= 25)                 // count quarters
    {
        amountLeft = amountLeft - 25;
        counter ++ ;
    }
    while (amountLeft >= 10)                 // count dimes
    {
        amountLeft = amountLeft - 10;
        counter ++ ;
    }
    while (amountLeft >= 5)                  // count nickels
    {
        amountLeft = amountLeft - 5;
        counter ++ ;
    }
    while (amountLeft >= 1)                  // count pennies
    {
        amountLeft = amountLeft - 1;
        counter ++ ;
    }
    printf("%i coins\n", counter);
}

File: pset6/cash/cash.py
from cs50 import get_float

counter = 0

# Prompt user for a positive input
while True:
    change = 100 * get_float("Change owed (in dollars): ")
    if change > 0:
        break
print(f"Cents:", change)

# Count quarters
while change >= 25:
    change = change - 25
    counter = counter + 1
    if change < 25:
        break
    
# Count dimes
while change >= 10:
    change = change - 10
    counter = counter + 1
    if change < 10:
        break
    
# Count nickels
while change >= 5:
    change = change - 5
    counter = counter + 1
    if change < 5:
        break

# Count pennies
while change >= 1:
    change = change - 1
    counter = counter + 1
    if change < 1:
        break
    
    
print(counter, "coin(s).")
        